### PR TITLE
Add auto-generated rockspec file to Lua API client

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

For https://github.com/swagger-api/swagger-codegen/issues/6284

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/lua/luarocks.mustache [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+dependencies = {
+	"lua >= 5.2"
+}
+
+build = {
+	type = "builtin",
+	modules = {
+		["http.request"] = "http/request.lua";
```
Comment (daurnimator) [inline modules/swagger-codegen/src/main/resources/lua/luarocks.mustache]: This section is meant to list the files you install. not the dependencies you need

File: modules/swagger-codegen/src/main/resources/lua/luarocks.mustache [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	summary = "API client genreated by Swagger Codegen",
+	detailed = [[
+{{{appDescription}}}]],
+	homepage = "https://github.com/swagger-api/swagger-codegen",
+	license = "Unlicense",
+	maintainer = "Swagger Codegen contributors",
+}
+
+dependencies = {
+	"lua >= 5.2"
```
Comment (daurnimator) [inline modules/swagger-codegen/src/main/resources/lua/luarocks.mustache]: Should add dependencies here. i.e. `http`

**Comment by wing328** [comment]
@daurnimator I made the suggested the change and rebased the PR on the latest master. Please review when you've time. Have a nice weekend.

**Comment by wing328** [comment]
`luarocks make --local petstore-1.0.0-1.rockspec` worked locally in my machine:
```
http 0.2-0 depends on lpeg_patterns >= 0.3 (not installed)
Installing https://luarocks.org/lpeg_patterns-0.4-0.all.rock
lpeg_patterns 0.4-0 is now installed in /home/dev01/.luarocks (license: MIT)

http 0.2-0 is now installed in /home/dev01/.luarocks (license: MIT)

petstore 1.0.0-1 is now installed in /home/dev01/.luarocks (license: Unlicense)
```

File: samples/client/petstore/lua/petstore-1.0.0-1.rockspec [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	"lua >= 5.2",
+	"http",
+	"dkjson",
+	"basexx"
+}
+
+build = {
+	type = "builtin",
+	modules = {
+		["pet_api"] = "petstore/api/pet_api.lua";
```
Comment (daurnimator) [inline samples/client/petstore/lua/petstore-1.0.0-1.rockspec]: This is not correct. It needs to be namespaced on the left. e.g. 
```
["petstore.api.pet_api"] = "petstore/api/pet_api.lua";
```

Comment (wing328) [reply samples/client/petstore/lua/petstore-1.0.0-1.rockspec]: Fixed via #7554

**Pull request merged**